Structure_Engine/Create/Elements/RetainingWall.cs

```diff
@@ -0,0 +1,116 @@
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using BH.oM.Base.Attributes;
+using BH.oM.Geometry;
+using BH.oM.Structure.Elements;
+using BH.oM.Structure.SurfaceProperties;
+using BH.Engine.Base;
+using BH.Engine.Geometry;
+using BH.Engine.Spatial;
+using BH.Engine.Structure;
+
+
+namespace BH.Engine.Structure
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Creates a RetainingWall from a Stem, Footing and properties.")]
+        [Input("stem", "The Stem of the RetainingWall.")]
+        [Input("footing", "The Footing of the RetainingWall.")]
+        [Input("retainedHeight", "The retained height of soil measured from the bottom of the wall Footing.")]
+        [Input("coverDepth", "The distance from top of Footing to finished floor level on the exposed face.")]
+        [Input("groundWaterDepth", "The distance from the base of the Footing to ground water level.")]
+        [Input("retentionAngle", "A property of the material being retained measured from the horizontal plane.")]
+        [Output("retainingWall", "The created RetainingWall containing the stem and footing.")]
+        public static RetainingWall RetainingWall(Stem stem, PadFoundation footing, double retainedHeight, double coverDepth, double retentionAngle, double groundWaterDepth = 0)
+        {
+            if (stem.IsNull() || footing.IsNull())
+                return null;
+
+            if (Query.IsValid(stem, footing))
+                return new RetainingWall() { Stem = stem, Footing = footing, RetainedHeight = retainedHeight, CoverDepth = coverDepth, RetentionAngle = retentionAngle, GroundWaterDepth = groundWaterDepth };
+            else
+                return null;
+        }
+
+        /***************************************************/
+
+        //TODO Add input descriptions here
```

Comment:
Complete this please, can you be clear that the line needs to exist in a single plane.